lib/src/Base/Geom/LevelSet.cxx

```diff
@@ -134,6 +139,26 @@ Bool LevelSet::contains(const Point & point) const
   return operator_(function_(point)[0], level_);
 }
 
+/* Compute the Euclidean distance from a given point to the LevelSet */
+Scalar LevelSet::computeDistanceToDomain(const Point & point) const
+{
+  if (contains(point)) return 0.0;
+  const UnsignedInteger dim(point.getDimension());
+  const TranslationFunction translate(point);
+  const ComposedFunction levelFunction(function_, translate);
+  const NearestPointProblem problem(levelFunction, level_);
+  AbdoRackwitz abdoRackwitz(problem);
+  Sample startingSample(2, dim); // AbdoRackwitz fails if levelFunction has null gradient at its starting point
```

Comment:
If a non-convex LevelSet is used, I guess that there is only a small chance of getting the minimum starting only with the points [0, ..., 0] and [1, ..., 1]. Would a random sample be useful?

Reply:
I guess, but how many sample points would we need? Which standard deviation do we need? Should we use an LHSExperiment?

Reply:
A robust way to compute this distance is to rely on `LevelSetMesher` then to build a `MeshDomain` using the resulting `Mesh` and you either return the result of `MeshDomain.computeDistance()` or use the nearest vertex to initialize your algorithm. It all depends on how robust you want to make this method and the price you are ready to pay for this robustness...

Reply:
@regislebrun How could we set the `boundingBox` argument of [LevelSetMesher.build](https://openturns.github.io/openturns/1.17/user_manual/_generated/openturns.LevelSetMesher.html#openturns.LevelSetMesher.build)?

Reply:
You can use the bounding box of the LevelSet!

Reply:
I don't think so. In the example code from the [LevelSetMesher doc page](https://openturns.github.io/openturns/1.17/user_manual/_generated/openturns.LevelSetMesher.html), the bounding box is manually specified:

```
import openturns as ot
mesher = ot.LevelSetMesher([5, 10])
level = 1.0
function = ot.SymbolicFunction(['x0', 'x1'], ['x0^2+x1^2'])
levelSet = ot.LevelSet(function, ot.LessOrEqual(), level)
mesh = mesher.build(levelSet, ot.Interval([-2.0]*2, [2.0]*2))
```
Moreover, ``levelSet.getLowerbound()``  and ``levelSet.getUpperbound()``both yield
``
class=Point name=Unnamed dimension=0 values=[]
``

Reply:
Nope. If you define a LevelSet object in the perspective of using it in this context, you have to define the bounds of the LevelSet. You can check the dimension of the bounds and throw the appropriate exception here. There is no magic here: you have to help the system a little bit...

Reply:
In the interest of getting this PR merged, I have removed the commit introducing LevelSet::computeDistance.